packages/cli/src/services/activeWorkflows.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Service } from 'typedi';
+
+import type { User } from '@db/entities/User';
+import { SharedWorkflowRepository } from '@db/repositories/sharedWorkflow.repository';
+import { WorkflowRepository } from '@db/repositories/workflow.repository';
+import { ActivationErrorsService } from '@/ActivationErrors.service';
+import { BadRequestError } from '@/errors/response-errors/bad-request.error';
+import { Logger } from '@/Logger';
+
+@Service()
+export class ActiveWorkflowsService {
+	constructor(
+		private readonly logger: Logger,
+		private readonly workflowRepository: WorkflowRepository,
+		private readonly sharedWorkflowRepository: SharedWorkflowRepository,
+		private readonly activationErrorsService: ActivationErrorsService,
+	) {}
+
+	async getAllActiveIds() {
+		const activationErrors = await this.activationErrorsService.getAll();
+		const activeWorkflowIds = await this.workflowRepository.getActiveIds();
+		return activeWorkflowIds.filter((workflowId) => !activationErrors[workflowId]);
+	}
+
+	async getAllActiveIdsForUser(user: User) {
+		const activationErrors = await this.activationErrorsService.getAll();
+		const activeWorkflowIds = await this.workflowRepository.getActiveIds();
+
+		const hasFullAccess = user.hasGlobalScope('workflow:list');
+		if (hasFullAccess) {
+			return activeWorkflowIds.filter((workflowId) => !activationErrors[workflowId]);
+		}
+
+		const sharedWorkflowIds =
+			await this.sharedWorkflowRepository.getSharedWorkflowIds(activeWorkflowIds);
+		return sharedWorkflowIds.filter((workflowId) => !activationErrors[workflowId]);
+	}
+
+	async getActivationError(workflowId: string, user: User) {
+		const hasAccess = await this.sharedWorkflowRepository.hasAccess(workflowId, user);
+		if (!hasAccess) {
+			this.logger.verbose('User attempted to access workflow errors without permissions', {
+				workflowId,
+				userId: user.id,
+			});
+
+			throw new BadRequestError(`Workflow with ID "${workflowId}" could not be found.`);
```

Comment:
For learning - I'd've used at `BadRequestError` and other response errors only at controller level. It's okay to use them at service level?

Reply:
I have conflicting opinions about this. one one hand all communication layer details should be abstracted away in the controllers, on the other hand letting services throw these error makes the code a lot simpler and easier to maintain 🤷🏽.